src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -204,12 +216,102 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
     }
 
     private static String lambdaClassName(Class<?> targetClass) {
+        return createNameFromTargetClass(targetClass) + counter.incrementAndGet();
+    }
+
+    private static String createNameFromTargetClass(Class<?> targetClass) {
         String name = targetClass.getName();
         if (targetClass.isHidden()) {
             // use the original class name
             name = name.replace('/', '_');
         }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda$";
+    }
+
+    /**
+     * Calculate hash value of the given String in the same manner as the
+     * java.lang.StringUTF16#hashCode does with difference that hash value
+     * is long instead of int.
+     *
+     * @param name String for which method calculates hash value for
+     *
+     * @return a hash value for the given String
+     * */
+    private String fixedSizeStringHash(String name) {
+        long h = 0;
+        int length = name.length();
+        for (int i = 0; i < length; i++) {
+            h = 31 * h + name.charAt(i);
+        }
+
+        StringBuilder hash = new StringBuilder(Long.toString(Math.abs(h), Character.MAX_RADIX));
+        int hashLength = hash.length();
+
+        // As all the hashes contained in the stable lambda names should
+        // be of the same length, we pad some of them with the character 'a'.
```

Comment:
Since `'a'` is part of the characters that are used for the encoding of the value itself it would be better to choose a character for padding that is never emitted by `Long.toString(hashValue, Character.MAX_RADIX)`. This might help debugging in the future.

Reply:
I like this idea. I used `#` since only small letters and digits are used for hashing. Also changed the comment accordingly.